### ISSUE TITLE
Unsized parameters are not flagged in trait definitions

### ISSUE THREAD

**Issue by ejmount** [rust-lang/rust#129252]
### Code

```Rust
trait Trait<T: ?Sized> {
    fn method(t: T); 
}
```

### Current output

```Shell
<none>
```

### Desired output

```Shell
A warning that the `?Sized` constraint is useless
```

### Rationale and extra context

Any concrete choice of `T` that's unsized will result in [E0277](https://doc.rust-lang.org/stable/error_codes/E0277.html) when the trait is implemented, so marking it as unsized isn't allowing anything that wasn't already valid. Similar redundant constructions like comparing unsigned values to 0 are flagged, so I'd expect this to be too. 

### Rust Version

```Shell
1.80.1
```
